TRIDENT-18367 Signal index migration FAQ (#2938)
SumoLogic/sumologic-documentation: docs/cse/records-signals-entities-insights/signal-index-migration-faq.md

---
id: signal-index-migration-faq
title: 2023 CSE Signal Index Migration FAQ
sidebar_label: Signal Index Migration FAQ
description: Learn about deprecation of the Signal Forwarding feature and what you need to know about migrating to the new sec_signal index.
---
<head>
  <meta name="robots" content="noindex" />
</head>

## Summary

> Starting last year, Signals generated by Cloud SIEM Enterprise (CSE) are automatically saved in a standardized `sec_signal` index. This special partition is similar to the existing `sec_record` indices in that, unlike data retained using the older Signal Forwarding feature, it is stored in a format that supports keyword search, nested attributes, and other standard log search features.

Reviewer: I'd specify `2022` in case this doc lives on into 2024 and beyond

```suggestion
Effective in 2022, Signals generated by Cloud SIEM Enterprise (CSE) are automatically saved in a standardized `sec_signal` index. This special partition is similar to the existing `sec_record` indices in that, unlike data retained using the older Signal Forwarding feature, it is stored in a format that supports keyword search, nested attributes, and other standard log search features.
```

     :::
     * Timestamps are now strings instead of integers.
     * The `prototype` attribute is now a string instead of a boolean.
     * The `extra` object has been deprecated. The `matchExpression` attribute contained in this object has been promoted to the top level of the log message.
* The following attributes/objects are no longer included in the new index:
     * `is_whitelisted` (which was deprecated in April 2022)
     * `multiEntitySharedId` (this was an internal attribute that was not meant to be exposed to users)
     * `new_rule_id` (duplicate of `rule_id`)
     * The `threat_indicators` object is no longer included (this has been deprecated for some time)
* Users may see a different number of signals for the same time period when comparing the `sec_signal` index with the old Signal Forwarding data. This is due to a number of factors related to how the Sumo platform and Cloud SIEM calculate start and end times and which timestamps are used in the different indices. But over the long term, both methods include the same signals.  

> The following table lists an example of the same Signal in the old format from Signal Forwarding and the new format in the `sec_signal` index:

Reviewer: ```suggestion
The following code snippet lists an example of the same Signal in the old format from Signal Forwarding and the new format in the `sec_signal` index:
```

> The following table lists an example of the same Signal in the old format from Signal Forwarding and the new format in the `sec_signal` index:

Reviewer: Should `table` be `code snippet `?

Author: Good catch, @kimsauce! I fixed it.

@pkazmir-sumo wrote the article with a table showing the code examples side-by-side. When I put his article into markdown, I placed the code examples one after the other, but forgot to change that sentence.

> The following table lists an example of the same Signal in the old format from Signal Forwarding and the new format in the `sec_signal` index:

Author: ```suggestion
The following code examples show the same Signal in the old format from Signal Forwarding and the new format in the `sec_signal` index:
```
